Backend/app/rider-platform/rider-app/Main/src/Storage/CachedQueries/Maps/DirectionsCache.hs

```diff
@@ -0,0 +1,49 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# OPTIONS_GHC -Wno-deprecations #-}
+
+module Storage.CachedQueries.Maps.DirectionsCache where
+
+import Data.Text
+import Domain.Types.Maps.DirectionsCache (DirectionsCache)
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import qualified Kernel.Storage.Hedis as Hedis
+import Storage.CachedQueries.CacheConfig (CacheFlow)
+import qualified Storage.Queries.Maps.DirectionsCache as Queries
+
+cacheDirectionsResponse :: CacheFlow m r => DirectionsCache -> m ()
+cacheDirectionsResponse dirCache = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  let dirKey = makeDirKey (dirCache.originHash) (dirCache.destHash) (dirCache.slot)
+  Hedis.setExp dirKey dirCache expTime
+
+makeDirKey :: Text -> Text -> Text -> Text
+makeDirKey originHash destHash slot = pack "CachedQueries:Maps:origin-" <> originHash <> "-dest:" <> destHash <> "-slot:" <> slot
+
+create :: DirectionsCache -> Esq.SqlDB ()
+create = Queries.create
+
+findRoute :: (CacheFlow m r, Esq.EsqDBFlow m r) => Text -> Text -> Text -> m (Maybe DirectionsCache)
+findRoute originHash destHash slot =
+  Hedis.safeGet (makeDirKey originHash destHash slot) >>= \case
+    Just x -> return $ Just x
+    Nothing -> do
```

Comment:
Why maybe return here?

Reply:
It is normal to not get the value that we are looking for in this table. This table is used to store paths from origin to destination and we may not have a path from an origin to destination stored (which basically implies we have to call google api and store its response for future use).